plugins/filter/filters.py

```diff
@@ -321,19 +336,29 @@ def client_properties(self, listener_dict, default_ssl_enabled, bouncy_castle_ke
                 final_dict[config_prefix + 'sasl.login.connect.timeout'] = '15000'
                 final_dict[config_prefix + 'sasl.oauthbearer.token.endpoint.url'] = oauth_token_uri
 
-                if oauth_groups_scope == 'none':
-                    final_dict[config_prefix + 'sasl.jaas.config'] = 'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \
-                    clientId=\"' + oauth_client_id + '\" clientSecret=\"' + str(oauth_client_password) + '\";'
+                if (oauth_groups_scope == 'none' and (not idp_self_signed)):
```

Comment:
This is also something similar to above one where there are 4 if else but can be reduced to 2. So either both shold be changed or both should be kept same

Reply:
yes, that's what I mentioned earlier about this file itself following this format for non oauth as well, ideally we should revamp the entire file to reduce the conditions but I think that's a separate task in itself which can not be prioritized right now. Something we can keep for Q3 improvements imo

Reply:
Sure lets keep it for Q3